### PR TITLE
bot, please handle mentions too, thanks

### PR DESCRIPTION
**Pull request by grzegorznowak**
**Proposed changes**:
- resolves directly inability to receive mentions , see:
https://github.com/RasaHQ/rasa_core/issues/1415

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible!

File: rasa_core/channels/slack.py [hunk 1]
```diff
@@ -135,6 +137,36 @@ def _is_button_reply(slack_event):
     def _get_button_reply(slack_event):
         return json.loads(slack_event['payload'][0])['actions'][0]['name']
 
+    @staticmethod
+    def _sanitize_user_message(text, uids_to_remove):
+        """
+        Probably a good starting point for pre-formatting of user-provided text,
+        to make NLUs life easier in case they go funky to the power of extreme.
```
Comment (ricwo) [inline rasa_core/channels/slack.py]: NLU's

File: rasa_core/channels/slack.py [hunk 2]
```diff
@@ -135,6 +137,36 @@ def _is_button_reply(slack_event):
     def _get_button_reply(slack_event):
         return json.loads(slack_event['payload'][0])['actions'][0]['name']
 
+    @staticmethod
+    def _sanitize_user_message(text, uids_to_remove):
+        """
```
Comment (ricwo) [inline rasa_core/channels/slack.py]: What does this function do? First line should be a one-line summary 
```text
"""One-line summary.

Details here(...)
""

File: rasa_core/channels/slack.py [hunk 3]
```diff
@@ -135,6 +137,36 @@ def _is_button_reply(slack_event):
...
+        Probably a good starting point for pre-formatting of user-provided text,
+        to make NLUs life easier in case they go funky to the power of extreme.
+
+        In the current state will just drop self-mentions of bot itself
+
+        :param text: raw message
+        :param users_to_drop: a list of users to remove from the content
+        :return text:
```
Comment (ricwo) [inline rasa_core/channels/slack.py]: Please check out the docstring convention used throughout Rasa Core, e.g. 
https://github.com/RasaHQ/rasa_core/blob/master/rasa_core/agent.py#L474

File: rasa_core/channels/slack.py [hunk 4]
```diff
@@ -135,6 +137,36 @@ def _is_button_reply(slack_event):
...
+        """
+        for uid_to_remove in uids_to_remove:
+            # the seeming duplication turns out to be an enough heuristic
```
Comment (ricwo) [inline rasa_core/channels/slack.py]: what do you mean by seeming duplication?

File: rasa_core/channels/slack.py [hunk 5]
```diff
@@ -135,6 +137,36 @@ def _is_button_reply(slack_event):
...
+            text = re.sub(
+                r'<@{}>\s'.format(uid_to_remove),
+                '',
+                text
+            )
+            text = re.sub(
+                r'\s<@{}>'.format(uid_to_remove),
+                '',
+                text
+            )
```
Comment (ricwo) [inline rasa_core/channels/slack.py]: what about cases like 
`<@123>hello`?

File: rasa_core/channels/slack.py (hunk 4 above)
Comment (ricwo) [inline rasa_core/channels/slack.py]: can we simplify this to something like
```python
for uid in uids_to_remove:
    for regex_string in [r'<@{}>\s'.format(uid),
                         r'\s<@{}>'.format(uid),
                         r'<@{}>'.format(uid)]:
        text = re.sub(regex_string, '', text)

return text.rstrip()
```

File: rasa_core/channels/slack.py [hunk 6]
```diff
@@ -178,7 +210,9 @@ def webhook():
                 elif self._is_user_message(output):
                     return self.process_message(
                         on_new_message,
-                        text=output['event']['text'],
+                        text=self._sanitize_user_message(
+                            output['event']['text'],
+                            output['authed_users']),
```
Comment (ricwo) [inline rasa_core/channels/slack.py]: I guess the bot itself is always part of `authed_users`?

File: rasa_core/channels/slack.py (hunk 5 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: we will tune further using test cases (see the comment below)

File: rasa_core/channels/slack.py (hunk 4 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: the duplication of regexp expressions

File: rasa_core/channels/slack.py (hunk 3 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: thanks yes will do when issuing the next changeset

File: rasa_core/channels/slack.py (hunk 2 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: thx, will add that

File: rasa_core/channels/slack.py (hunk 1 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: thx!

File: rasa_core/channels/slack.py (hunk 3 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: hey @ricwo and what's the convention for Return values in docstring ? I couldn't find any example in the linked file. If None I would just:
`Returns text: a parsed and cleaned string`

**Comment by grzegorznowak** [comment]
Hey @ricwo . Docstrings updated, code reformatted slightly, and most importantly the sanitizer's tests added. That should hopefully cover all the suggestions above. Let me know

File: rasa_core/channels/slack.py (hunk 4 above)
Comment (grzegorznowak) [reply rasa_core/channels/slack.py]: have tuned that comment to not confuse people

File: rasa_core/channels/slack.py [hunk 7]
```diff
@@ -135,6 +137,32 @@ def _is_button_reply(slack_event):
...
+        Probably a good starting point for pre-formatting of user-provided text,
+        to make NLU's life easier in case they go funky to the power of extreme.
+
+        In the current state will just drop self-mentions of bot itself
+
+        Args:
+            text: raw message as sent from slack
+            uids_to_remove: a list of user ids to remove from the content
+
+        Returns text: a parsed and cleaned string
```
Comment (ricwo) [inline rasa_core/channels/slack.py]:
```suggestion
        Returns:
            str: parsed and cleaned version of the input text 
```

**Comment by grzegorznowak** [comment]
hey, thanks for the tweaks and prompt responses @ricwo, that was legendary quick.  Cheers

**Comment by grzegorznowak** [comment]
hey @ricwo sorry is there anything else for me to do here ? Not sure if the current state concludes the PR in your eyes.

**Comment by grzegorznowak** [comment]
done @ricwo

**Comment by grzegorznowak** [comment]
sorry @ricwo  the changelog should also include this:
```
SlackInput wont ignore `app_mention` event anymore; so will handle messages containing @mentions and will respond to these (as long as the event itself is enabled in the application hosting the bot)
```
let me know should I make another pr or can you author that directly into `master`. Sorry for the trouble!

**Comment by ricwo** [comment]
@grzegorznowak yes, please create another PR

**Pull request merged**